From 7b9157bca7138480c387ef3d5b69b6cf1eb498e5 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Fri, 15 Dec 2017 22:45:55 -0500
Subject: [PATCH] CUDA 9 compatible gpu-architecture default

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)


Copyright notice for CMakeLists.txt:

#
#   Copyright 2013 Pixar
#
#   Licensed under the Apache License, Version 2.0 (the "Apache License")
#   with the following modification; you may not use this file except in
#   compliance with the Apache License and the following modification to it:
#   Section 6. Trademarks. is deleted and replaced with:
#
#   6. Trademarks. This License does not grant permission to use the trade
#      names, trademarks, service marks, or product names of the Licensor
#      and its affiliates, except as required to comply with Section 4(c) of
#      the License and to reproduce the content of the NOTICE file.
#
#   You may obtain a copy of the Apache License at
#
#       http://www.apache.org/licenses/LICENSE-2.0
#
#   Unless required by applicable law or agreed to in writing, software
#   distributed under the Apache License with the above modification is
#   distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
#   KIND, either express or implied. See the Apache License for the specific
#   language governing permissions and limitations under the Apache License.
#

# This patch was obtained from the gentoo-overlay

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4f3cd9d4..fa438b46 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -513,8 +513,10 @@ if(CUDA_FOUND)
         if (NOT DEFINED OSD_CUDA_NVCC_FLAGS)
             if (CUDA_VERSION_MAJOR LESS 6)
                 set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_11 )
-            else()
+            elseif (CUDA_VERSION_MAJOR LESS 9)
                 set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_20 )
+            else()
+                set( OSD_CUDA_NVCC_FLAGS --gpu-architecture compute_30 )
             endif()
         endif()
     endif()
